doc/modules/compose.rst

```diff
@@ -251,12 +251,12 @@ the regressor that will be used for prediction, and the transformer that will
 be applied to the target variable::
 
   >>> import numpy as np
-  >>> from sklearn.datasets import load_boston
+  >>> from sklearn import datasets
   >>> from sklearn.compose import TransformedTargetRegressor
   >>> from sklearn.preprocessing import QuantileTransformer
   >>> from sklearn.linear_model import LinearRegression
   >>> from sklearn.model_selection import train_test_split
-  >>> X, y = load_boston(return_X_y=True)
+  >>> X, y = datasets.fetch_openml('boston', return_X_y=True)
```

Comment:
oh, I totally forgot about these. Very good point. We should probably change them to another dataset, and not have boston in our examples at all.

Reply:
We have loads of examples with boston dataset. I don't think I would be able to judge if another dataset is suitable for given example

Reply:
Yes, I think we can do this in multiple steps. First replace with boston from openml and deprecate `load_boston` then see if we want to replace it with some other dataset (which will require significant work to adapt examples).

Reply:
Should we do it the other way around?

First open many pull requests to stop using the boston dataset in our education material (examples and documentation).

Then once all those changes are merged, deprecate `load_boston` dataset.

We should not encourage people to use `fetch_openmlp("boston")`.

I think we should strive to be exemplary ourselves instead than asking other people to update their code before we do.

Reply:
Indeed, but to address it in this sprint, there should be a consensus what datasets we use to replace it in the parent issue https://github.com/scikit-learn/scikit-learn/issues/16155 (e.g. California or Ames housing). So far it didn't look like there was a clear decision about it. I could be decided on a case by case basis I imagine.

California housing is 40x bigger, and would be too large for a number of cases where boston is currently used.

Reply:
As discussed IRL:

- in tests and class docstrings we can use synthetic data from `sklearn.datasets.make_regression`.
- in most examples we can probably use `fetch_california_housing` and possibly subsample it with `train_test_split(X, y, train_size=1000, test_size=300, random_state=0)` with a comment that explains that we subsample the training set to keep the example runtime low.

Reply:
Some examples will require significant work such as:

```
examples/plot_partial_dependence_visualization_api.py
```